app/models/order.rb

```diff
@@ -1,19 +1,34 @@
 class Order < ApplicationRecord
 
   enum status: [:pending, :packaged, :shipped, :cancelled]
-  #validates_numericality_of :status
 
   belongs_to :user
   has_many :order_items
   has_many :items, through: :order_items
 
 
   def item_quantity
-    items.count
+    order_items.sum :quantity
   end
 
   def items_total_value
-    items.sum(:price)
+    order_items.sum { | o_i| o_i.price * o_i.quantity }
   end
 
+  def items_from_merchant(merchant_id)
+    items.where(user_id: merchant_id)
+  end
+
+  def item_count_for_merchant(merchant_id)
+    items_from_merchant(merchant_id).sum do |item|
+      item.order_items.find_by(order_id: id).quantity
+    end
+  end
+
+  def item_total_value_for_merchant(merchant_id)
```

Comment:
I hope there is an easier way to calculate only a merchants items in an order. Open to any/all suggestions :)